### PR TITLE
Fix itty.hasNext() and commit invoking disordering

### PR DESCRIPTION
**Pull request by sandszhouSZ**
Refer:
 https://issues.apache.org/jira/browse/TINKERPOP-2257?filter=-2

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -35,6 +35,9 @@ This release also includes changes from <<release-3-3-8, 3.3.8>>.
 * Bump to Netty 4.1.36.
 * Bump to Groovy 2.5.7.
 
+==== Bugs
```
Comment (spmallette) [inline CHANGELOG.asciidoc]: There is no need to add this manually - we add those on release. You should however add a manual changelog entry above that describes your change.

**Comment by spmallette** [comment]
Thank you for this. I assume this problem goes back to 3.3.x - if so, could you please target the `tp33` branch instead.

**Comment by sandszhouSZ** [comment]
> Thank you for this. I assume this problem goes back to 3.3.x - if so, could you please target the `tp33` branch instead.

I review the branch list， and util 3.3.7, the relate code seems ok:
image
In fact，I found this error occured since branch: 3.4.0-rc2

**Comment by spmallette** [comment]
ok - metadata must have come in 3.4.0. we didn't have the feature in 3.3.x i guess.

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (sandszhouSZ) [reply CHANGELOG.asciidoc]: Ok, I have removed changes in  CHANGELOG.asciidoc

Comment (sandszhouSZ) [reply CHANGELOG.asciidoc]: python test fail, and i thinks it's travis-ci problem also

**Comment by sandszhouSZ** [comment]
python test fail, and i thinks it's  travis-ci problem

**Comment by spmallette** [comment]
For other reviewers, I will add a CHANGELOG entry on merge.

VOTE +1

**Comment by divijvaidya** [comment]
VOTE +1 

The logic and the code change looks good.

**Comment by spmallette** [comment]
Added changelog entry via CTR : https://github.com/apache/tinkerpop/commit/0f95e00574171a9648a2bdd82c8867e9e87b00d6

**Pull request merged**